File: 1441. Build an Array With Stack Operations/Solution.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        ls = []
        ptr = 0
        for i in range(1,n+1):
            if ptr>=len(target):
                break
            if i==target[ptr]:
                ls.append("Push")
                ptr+=1
            else:
                ls+=["Push","Pop"]
        return ls

File: 1441. Build an Array With Stack Operations/Solution.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> ret = new ArrayList<>();
        int ptr = 0;
        int k=1;
        while(ptr<target.length){
            if(target[ptr]==k){
                ret.add("Push");
                ptr+=1;
            }else{
                ret.add("Push");
                ret.add("Pop");
            }
            k+=1;
        }
        return ret;
    }
}